### DISCUSSION TITLE
ViaVersion/ViaBackwards equivalent for itzg vanilla docker bedrock server?

### DISCUSSION THREAD

**Discussion by stembus** [itzg/docker-minecraft-bedrock-server#445] [General]
Hello there,

firstly, sorry for my dump question, I am new to minecraft server "world" - father of a boy who wants to have his own server and play together with local friends! :)

I got ITZG Docker up and running on my Synology through Container Manager and Docker compose - it works like charm and thanks @itzg I will definitely send you some coffee ;) 

However, now I wanted to test some of my sons clients and all I am getting is "outdated client" issue - I googled for some solutions and what I found was to use some plugins like ViaVersion/ViaBackwards etc... but I also found out that plugins are not supported on Vanilla servers so... I wonder whether you all always runs single/same version of server and all the clients (impossible for me in real life if I want sons friends from school to play there) or... am I missing something? 

Is there a way to allow older clients to the server via some "feature"?

Thanks a lot
Jiri

**Comment by itzg** [comment]
You will actually need to run a Java edition server and add Bedrock compatibility plugins and then can add ViaVersion etc

https://docker-minecraft-server.readthedocs.io/en/latest/misc/examples/#bedrock-compatible-server

Otherwise "outdated client" can often be solved by checking the App Store / Play store for updates tot he client.

**Reply by stembus** [reply]
Hello Geoff - thanks a lot for your swift reply - appreciate! ;)

Do you have such experience with running Java version for Bedrock clients? Does it work? you know... its for a kids, want to have it 24/7 on Synology without extra "care" and ... unfortunately forcing all the clients (which I dont have under control) to the latest versions is... impossible, some of them use old HW Android/IOS versions that do not support latest versions, some use switch consoles etc... so... I am looking for some "ultimate" solution that will work and brings joy rather than frustration for the kids :)

anyway I will give it a try with your java docker and will see, should be pretty straight forward at least to set it up...

**Reply by DominusExult** [reply]
I‘ve been doing that for my kids (java server + via plugins).
The only downside is that the java server is likely to update automatically and the plugins won’t (and likely won’t have a compatible version for a few days). So your server will crash until you manually update the plugins 🤦‍♂️

**Reply by stembus** [reply]
but java server will update only when you restart the docker and there is a newer version available, right? Means... if you dont restart (dont allow auto-start etc) then you still should have it "somehow" under manual control - right?

**Reply by DominusExult** [reply]
That‘s true 🤪
But there is another useful Docker thing that shuts down the server when no one is connected for x time and restarts it when someone connects (this makes the first connection fail but I hated having the server idling for hours when the kids and their friends were in bed and in school).
I could also disable autoupdate 🤷‍♂️

**Reply by itzg** [reply]
Be sure to set `VERSION` and  it won't surprise you with an update at all

https://docker-minecraft-server.readthedocs.io/en/latest/versions/minecraft/

Same goes for most of the plugin download features with version pinning.

**Comment by FalkRaft** [comment]
There's a mod or proxy called ViaBedrock and a client-side mod called ViaFabricPlus builds on top of ViaBedrock. ViaBedrock can translate Java packets into Bedrock packets, but you may run into some issues, some being:
- Client-authoritative inventory not working; you need to use /give or /replaceitem to change your inventory.
- Custom commands - some custom commands may override Java's chat commands and not load them on the client.
- Permission issues - some commands may not work.
- Movement works, but will lead to rubberbanding.
- Common edition differences, e.g. Bedrock has its own bridging mechanics, Java's shorter block reach.
